playbooks/tasks/portal-role-task-mongo-shell-eval.yml

```diff
@@ -6,9 +6,8 @@
 - name: Run mongo shell command
   community.mongodb.mongodb_shell:
     # Use secure localhost connection unless on subcrap test cluster
-    login_host: "{{ 'localhost' if (subcrap_mongo_port is not defined) else ansible_host }}"
-    # Use default port unless on subcrap test cluster
-    login_port: "{{ subcrap_mongo_port | default(default_mongo_port | string) }}"
+    login_host: "{{ 'localhost' if (custom_mongo_port is not defined) else ansible_host }}"
```

Comment:
I do not understand the logic here, why does `custom_mongo_port` affect the host ?

Reply:
I still don't get it I think. Why would we use localhost instead of ansible_host, it doesn't seem like a valid setup case.

Reply:
If we don't need to test mongo on virtual machines there would be only:
`login_host: "localhost"`

The `localhost` in this task means a mongo container under which we run the mongo shell script (`mongo_cmd: "docker exec mongo mongo"`).

With `localhost` we are sure that mongo setup scripts stay and don't in any case leave/leak out of mongo container.

When there is a need to test mongo cluster which is setup e.g. on `crap` host machine as `subcrap` virtual machines we can loose up the security a bit and target `subcrap` mongo nodes we are using `ansible_host` (instead of `localhost` which targets `crap` mongo node).

OTOH there was a coupling between using custom mongo port and using non-localhost target, i.e. using custom mongo port assumed subcrap testing of mongo cluster. I decoupled it so that custom mongo port is now independent of mongo cluster tests via usage of `use_custom_mongo_host` variable.